### PR TITLE
Update DG: More Relevant Sections and Model Component UML

### PR DESCRIPTION
**Pull request by zikunz**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -11,16 +11,31 @@
...
-    - [2.7. Storage Component](#27-storage-component)
-    - [2.8. Common Component](#28-common-component)
+    - [2.1. Architecture](#2.1-architecture)
+    - [2.2. EasyLog Component](#2.2-easylog-component)
+    - [2.3. UI Component](#2.3-ui-component)
+    - [2.4. Parser Component](#2.4-parser-component)
+    - [2.5. Command Component](#2.5-command-component)
+    - [2.6. Model Component](#2.6-model-component)
+    - [2.7. Storage Component](#2.7-storage-component)
+    - [2.8. Common Component](#2.8-common-component)
```
Comment (ongweisheng) [inline docs/DeveloperGuide.md]: undo this part, hyperlink is not working anymore

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -11,16 +11,31 @@
...
+    - [6.2. Continuous Integration](#continuous-integration)<br>
+    - [6.3. Coverage Report](#coverage-report)<br>
+    - [6.4. Making a Release](#making-a-release)<br>
+    - [6.5. Managing Dependencies](#managing-dependencies)
+- [Appendices](#appendices)
+    - [Appendix A: Product Scope](#appendix-a-product-scope)
+    - [Appendix B: User Stories](#appendix-b-user-stories)
+    - [Appendix C: Non-Functional Requirements](#appendix-c-non-functional-requirements)
+    - [Appendix D: Glossary](#appendix-d-glossary)
+
```
Comment (ongweisheng) [inline docs/DeveloperGuide.md]: not sure if this part is required, might delete later

**Review by ongweisheng — CHANGES_REQUESTED** [review]

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (zikunz) [reply docs/DeveloperGuide.md]: Sorry, I will make all hyperlinks following the convention we did for UG.

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (zikunz) [reply docs/DeveloperGuide.md]: We could keep for now. If not required, we can delete later :)

**Review by ongweisheng — APPROVED** [review]

**Pull request merged**